### DISCUSSION TITLE
v0.9.0

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#363] [Announcements]
此版本要求 [.NET 6.0.4 Desktop 运行时](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.4-windows-x64-installer)以及 DirectX Feature Level 11。

**新功能**

1. 全面重写渲染系统，新的 MagpieFX v2，语法见[自定义效果](https://github.com/Blinue/Magpie/wiki/%E8%87%AA%E5%AE%9A%E4%B9%89%E6%95%88%E6%9E%9C%EF%BC%88MagpieFX%EF%BC%89)
2. 实现游戏内覆盖，目前仅用于衡量性能，以后也会用于在运行时更改效果参数。默认快捷键 Ctrl+Alt+D，可在设置中更改。
3. 拖拽 FPS 可调整位置，在 FPS 上右键可调整样式 #341
4. 增强了光标功能，现在可以将光标自然的移到其他窗口上，如开始菜单、Xbox Game Bar 和 Alt+Tab 窗口
5. 在 ScaleModels.json 中可以为效果指定 inlineParams 参数以启用内联参数。这可以稍微提高某些效果的性能，但每次更改参数时都需要重新编译该效果。

示例：
``` json
{
    "effect": "FSR_RCAS",
    "sharpness": 0.87,
    "inlineParams": true
}
```
6.  “在 3D 游戏中限制光标”已升级为“3D 游戏模式”，启用时将针对 3D 游戏进行优化，游戏内覆盖也有不同的表现。
7. 调整缩放模式预设，移除 RAVU，Lanczos 不再包含锐化步骤。

**效果更改**

1. 所有效果均已重写。性能对比见 https://github.com/Blinue/Magpie/discussions/254#discussioncomment-2251071
2. 添加 CatmullRom，这是 Bicubic 的一个变体，针对速度优化，最适合下采样
3. 添加 CAS_Scaling，支持缩放的 CAS 变体
4. 添加 LCAS，一个非常轻量的 3D 画面缩放算法（感谢 @Tzugimaa）
5. 添加 MMPX，一个像素画缩放算法 #349
6. 添加 NNEDI3_nns16_win8x4 和 NNEDI3_nns64_win8x6 #283
7. 删除 Anime4K_Upscale_S_Lite、Bicubic_Lite 和 FSR_EASU_DX10
8. Linear 重命名为 Bilinear
9. AdaptiveSharpen 的 curveHeight 参数默认值调整为 0.8
10. 移除了 CRT_Easymode 的 scaleInLinearGamma 参数
11. SSimDownscaler 添加 oversharp 参数用于调整锐度

**系统需求**
1. Windows 10/11
2. DirectX Feature Level 11

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.9.0'>v0.9.0</a>.</em>

**Comment by SpezFishz** [comment]
什么时候Magpie支持FSR2.0

**Reply by Blinue** [reply]
https://github.com/Blinue/Magpie/discussions/284#discussioncomment-2400536

**Reply by YorkFan4925** [reply]
考虑适配arm架构pd虚拟机吗？m1在虚拟机上打不开

**Reply by Blinue** [reply]
#308

**Reply by Blinue** [reply]
@zxyyangxz

**Reply by Blinue** [reply]
@YorkFan4925

**Reply by Blinue** [reply]
关于适配 ARM 的问题请统一在 #308 中讨论

**Reply by YorkFan4925** [reply]
刚刚看到回复了，谢谢大佬！我以为没明白的那个是开发者

**Comment by zxyyangxz** [comment]
大神，我是玩游戏的小白。抱歉，这句话属实没看懂。



在 2022-05-31 20:54:49，"YorkFan4925" ***@***.***> 写道：

考虑适配arm架构pd虚拟机吗？m1在虚拟机上打不开

**Comment by YorkFan4925** [comment]
我是用MacBook m1安装x86
arm架构win11系统的，一般是x64系统的嘛，但是您的软件打不开，但是没有报错，就想着应该是没有适配，想请问作者有考虑适配吗？

**Comment by yutouyes** [comment]
Edit: 问题解决，是我搞反显示器了，非常抱歉。。 

请问一下magpie里的垂直同步是以最低刷新率的显示器为基准吗，我的主屏幕是151Hz，副屏是148Hz

![image](https://user-images.githubusercontent.com/17039457/178080421-543ff84f-2860-49c6-9ec7-f9c1e84d61a3.png)
![image](https://user-images.githubusercontent.com/17039457/178080411-7190f8dd-abfc-4789-bbe8-cb555d19ad01.png)

在主屏上使用magpie，垂直同步的帧率是148FPS，这个是否可以修改
![image](https://user-images.githubusercontent.com/17039457/178080469-c6caa473-a1dc-42d9-b46e-83e0d3d801ec.png)
![T7@HFH80V )JHZ$T~Z$(4MH](https://user-images.githubusercontent.com/17039457/178080398-71fbc442-1ea8-474c-91b2-fbda35c6357d.png)

另外想请问一下0.9的帧数限制选项在哪，现在关闭垂直同步之后似乎是无限制帧数了，而且与RTSS显示的FPS不一致
![ZDFK8~UQ3_(@R)N$ 0 ~24](https://user-images.githubusercontent.com/17039457/178080519-bf6a0e9d-a3d6-4438-a372-d9999fc648fb.png)

**Reply by hooke007** [reply]
> 0.9的帧数限制选项

dup https://github.com/Blinue/Magpie/issues/378#issuecomment-1140141711

> 与RTSS显示的FPS不一致

https://github.com/Blinue/Magpie/wiki/FAQ#magpie-%E4%B8%AD%E6%98%BE%E7%A4%BA%E7%9A%84%E5%B8%A7%E7%8E%87%E6%98%AF%E4%BB%80%E4%B9%88
